### PR TITLE
Preliminary 1.17 work

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to get Skript ready for MC 1.17

### NOTES:
- Particles -> I haven't used TPGames' new particle system yet, so all particles are added as normal particles, but the vibration and dust_color_transition particles will need some more work in the code
- From what I can see in the Bukkit 1.17 update, there are no potions/enchants/anything else that need to be added. Please let me know if I missed anything
- Thanks to @TPGamesNL for updating Gradle and dealing some other minor changes

---
**Target Minecraft Versions:** 1.17
**Requirements:** 1.17 Paper API
**Related Issues:** none

### CODE REVIEW

File: src/main/resources/lang/english.lang [hunk 1]
```diff
@@ -394,6 +394,9 @@ biomes:
 	soul_sand_valley: soul sand valley
```
Comment (APickledWalrus) [inline src/main/resources/lang/english.lang:394]: not sure where to put this so I'm just putting it here lol

I think you still need to add the composter inventory type

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
just the one thing

File: src/main/java/ch/njol/skript/entity/AxolotlData.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	@Nullable
+	private Variant variant = null;
+
+	public AxolotlData() {}
+
+	public AxolotlData(@Nullable Variant variant) {
+		this.variant = variant;
+		matchedPattern = variant != null ? variant.ordinal() + 1 : 0; // is this right?
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/entity/AxolotlData.java]: thanks. I tested it a decent amount and it seemed to work, then I forgot to remove the note.
So thanks for the reminder

File: src/main/resources/lang/english.lang (hunk 1 above)
Comment (ShaneBeee) [reply src/main/resources/lang/english.lang:394]: added, thanks for the remind

**Review by APickledWalrus — APPROVED** [review]
Looks good to me. I suppose the particle changes should be revised by TP maybe just incase.

**Comment by ShaneBeee** [comment]
> Looks good to me. I suppose the particle changes should be revised by TP maybe just incase.

thats what I was thinking. I didnt want to get into that right now, as I haven't spent time with his new system.

**Comment by TPGamesNL** [comment]
The particle stuff looks good, but indeed `DUST_COLOR_TRANSITION` and `VIBRATION` need extra data, I'll add that

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**